soc/nxp/mcx/mcxw/Kconfig.defconfig

```diff
@@ -49,13 +49,13 @@ config HCI_NXP_RX_THREAD
 
 endif # BT
 
-if NET_L2_IEEE802154 || NET_L2_OPENTHREAD
+if IEEE802154
```

Comment:
The symbols are meant for different purposes
* `CONFIG_NET_L2_IEEE802154` and `CONFIG_NET_L2_OPENTHREAD` are set if IEEE 802.15.4 L2 layer is enabled
* `CONFIG_IEEE80154` is set if IEEE 802.15.4 driver support is enabled. One can have driver support enabled without L2 support or have a custom L2 support in which case the configs in previous line are not enabled

Not sure what is the purpose of this change, but at first glance it does not look it is really needed.

Reply:
Thank you.
It was wrong to use the NET_L2 macros to begin with. NXP_RF_IMU enables the communication with the RF core so it should be guarded by a lower level driver define as CONFIG_IEEE80154.